recipes/cgns/all/conanfile.py

```diff
@@ -37,23 +40,37 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def validate(self):
+        if self.options.parallel and not self.options.with_hdf5:
+            raise ConanInvalidConfiguration("Parallel requires HDF5 with parallel enabled")
+
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
+        if self.options.parallel:
+            self.options["hdf5"].parallel = True
+            self.options["hdf5"].enable_cxx = False # can't enable this with parallel
```

Comment:
this would need to be in the validate as well

Reply:
As well, or instead-of ?
Should this recipe be setting hdf5's options like this?
Or should we simply validate to check the consumers have set the appropriate options in HDF5 ?

Reply:
in general, it's not recommended to modify optiions like that
instead, invalid combinations are rejected in `validate`
but sometimes it's not possible (e.g. because of necessity to use `shared` dep while `static` is default one)

Reply:
I assume the method I use now is ok?